internal/controller/kuik/cachedimage_controller.go

```diff
@@ -343,7 +344,31 @@ func (r *CachedImageReconciler) cacheImage(cachedImage *kuikv1alpha1.CachedImage
 		return err
 	}
 
-	err = registry.CacheImage(cachedImage.Spec.SourceImage, desc, r.Architectures)
+	lastUpdateTime := time.Now()
+	lastWriteComplete := int64(0)
+	onUpdated := func(update v1.Update) {
+		needUpdate := false
+		if lastWriteComplete != update.Complete && update.Complete == update.Total {
+			// Update is needed whenever the writing complmetes.
+			needUpdate = true
+		}
+
+		if time.Since(lastUpdateTime).Seconds() >= 5 {
+			// Update is needed if last update is more than 5 seconds ago
+			needUpdate = true
+		}
+		if needUpdate {
```

Comment:
We are not really in control when and how often the `WithProgress()` callback will be called. They are handled by `remote` package. 

The purpose of `lastWriteComplete != update.Complete` is to make sure that we don't unnecessarily update the status blindly multiple times if in case the call back was called multiple times with same progress value.

Reply:
I have applied the code change as suggested.